test/Main.hs

```diff
@@ -316,6 +316,26 @@ autocommitTest td = testCase "Autocommit mode works" .
     assertEqualEq "Other connection sees autocommited data" 1 n
   runQuery_ $ rawSQL "DELETE FROM test1_ WHERE a = $1" sint
 
+preparedStatementTest :: TestData -> Test
+preparedStatementTest td = testCase "Execution of prepared statements works" .
+                           runTestEnv td defaultTransactionSettings $ do
+  let i1 = 42 :: Int32
+  runPreparedQuery_ "select1" $ "SELECT" <?> i1
+  o1 <- fetchOne runIdentity
+  assertEqualEq "Results match" i1 o1
```

Comment:
Shouldn't there be some sort of a check that verifies that the prepared statement actually exists in the database and is prepared? I'm not sure if that's even possible.

Reply:
https://www.postgresql.org/docs/11/view-pg-prepared-statements.html ?

Reply:
Added.